flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/CalcITCase.scala

```diff
@@ -392,6 +392,34 @@ class CalcITCase extends BatchTestBase {
         Timestamp.valueOf("2019-09-19 08:03:09").toInstant)))
   }
 
+  @Test
+  def testTimeUDFParametersImplicitCast(): Unit = {
+    val data = Seq(row(
+      localDateTime("2019-09-19 08:03:09"),
+      Timestamp.valueOf("2019-09-19 08:03:09"),
+      Timestamp.valueOf("2019-09-19 08:03:09").toInstant))
+    registerCollection("MyTable", data,
+      new RowTypeInfo(LOCAL_DATE_TIME, TIMESTAMP, INSTANT),
+      "a, b, c")
```

Comment:
Would be better to use DDL to register tables which can explicitly define sql types. We can also add different precision columns, e.g. 

```scala
    val ddl =
      """
        |CREATE TABLE src (
        |  ntz TIMESTAMP(3),
        |  ltz0 TIMESTAMP_LTZ(0),
        |  ltz3 TIMESTAMP_LTZ(3),
        |  ltz6 TIMESTAMP_LTZ(6),
        |  ltz9 TIMESTAMP_LTZ(9),
        |  ltz_not_null TIMESTAMP_LTZ(3) NOT NULL
        |) WITH (
        |  'connector' = 'values',
        |  ...
        |)
        |""".stripMargin
```